src/Host/Client/Impl/Session/RSessionProvider.cs

```diff
@@ -50,6 +51,12 @@ public class RSessionProvider : IRSessionProvider {
             }
         }
 
+        public Task RestartSessions(IRHostBrokerConnector brokerConnector) {
+            var sessions = _sessions.Values.Where(s => s.BrokerConnector.Equals(brokerConnector)).ToList();
```

Comment:
That's a good question. `BrokerID` belongs to the broker (broker's URL), but theoretically we may want to have several broker connectors that work with the local broker. So I think we should compare the instance itself rather than its broker ID

---
In reply to: [73031834](https://github.com/Microsoft/RTVS/pull/2133#discussion_r73031834) [](ancestors = 73031834)